README.md

```diff
@@ -43,17 +57,127 @@ Snippets for deployment template scaffolding and 70+ Azure resources.
+
+![Image showing a schema validation error](./images/schema-validation.png)
+
+## Parameter files
+
+Associating a parameter file with an ARM template enables full validation across both the parameter file and template. A parameter file association can be made by either creating a new parameter file from an existing template or by associating an existing parameter file to the template.
+
+### Create a new parameter file
+
+To create a new parameter file from an existing template, click on the code lense text `**Select or create a parameter file to enable full validation..**.
+
+![Image showing the code lense for creating a parameter file association](./images/create-parameter-file.png)
+
+Select **New** from the context menu.
+
+![Image showing the parameter file creation options (none, new, and browse)](./images/create-parameter-file-2.png)
+
+Selecting **All parameters** adds every parameter from the template into the new parameter file. Selecting **Only required parameters** adds only parameters that do not have a default value specified in the template file.
+
+![Image showing options for adding parameters for parameter file (All parameters, only required parameters)](./images/create-parameter-file-3.png)
+
+Select a name and location for the parameter file.
+
+![Image showing a file save as dialog box](./images/create-parameter-file-4.png)
+
+Once done, a parameter file has been created. Where needed, remove the comment indicating that a parameter value is needed and add a parameter value.
+
+![Image showing new parameter file](./images/parameter-file.png)
+
+Also, notice that on the template file that the code lense text has been updated to reflect the current parameter file association.
+
+![Image showing that the parameter code lens has been updated to reflect associated parameter file](./images/code-lense.png)
+
+### Associate an existing parameter file
+
+To create an association with an existing parameter file, click on the code lense text `**Select or create a parameter file to enable full validation..**.
+
+![Image showing the code lense for creating a parameter file association](./images/create-parameter-file.png)
+
+Select **Browse** from the context menu and then select the parameter file.
+
+![Image showing the parameter file creation options (none, new, and browse)](./images/create-parameter-file-2.png)
+
+### Parameter validation
+
+Once a parameter file has been associated with a template, the extension provides parameter validation across both files. In the following example, a parameter is specified with a set of allowed values. In the parameter file, a value is specified that is not part of the allowed values list. Because of this, the extension produces an error.
+
+![Image showing a parameter validation issue](./images/param-file-validation.png)
```

Comment:
Very ugly error from validation, do we have a simpler example?

Reply:
This error is demonstrating that a value has been specified in the parameters file that is not part of the allowed values list. I've looked at a validation error for a value that exceeds max value and the error is equally as ugly. I'm not sure what parameter validation would have a more simple error, if you can provide one, I will update.